src/components/WalletList.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import './WalletList.css';
+
+type WalletEntryParams = {
+  name: string;
+  uuid: string;
+  rdns: string;
+  icon: string;
+  extensionId?: string;
+};
+
+export type WalletMapEntry = {
+  params: WalletEntryParams;
+  eventName: string;
+};
+
+type WalletListProps = {
+  wallets: Record<string, WalletMapEntry>;
+};
+
+const WalletList: React.FC<WalletListProps> = ({ wallets }) => {
+  const walletEntries = Object.entries(wallets);
+
+  if (!walletEntries.length) {
+    return null;
+  }
+
+  return (
+    <div className="wallet-list">
+      {walletEntries.map(([key, wallet]) => (
+        <div key={key} className="wallet-card">
+          <img
+            src={wallet.params.icon}
+            alt={`${wallet.params.name} icon`}
+            className="wallet-icon"
+          />
```

Comment:
not a blocker but would be cool if each of these cards had a button where if clicked you'd connect to that wallet. For now we can/should keep it to one wallet connection at a time so when this happens perhaps we could set the extensionId in the input field and disable the input.

Reply:
fixed in : [aff4bcdd](https://github.com/MetaMask/test-dapp-multichain/pull/22/commits/aff4bcdda17338e33d694857ba99cbc3132f3935)

new card
<img width="445" alt="Screenshot 2024-12-11 at 11 04 04" src="https://github.com/user-attachments/assets/ef6b353b-6c34-455d-a084-5ec588df8dc7">